docs/oid-info.md

```diff
@@ -2,52 +2,89 @@
 
 ## Description
 
-Sigstore maintains its own Private Enterprise Number (57264) with the Internet
+Sigstore maintains its own Private Enterprise Number ([57264](http://oid-info.com/get/1.3.6.1.4.1.57264)) with the Internet
 Assigned Numbers Authority to help identify and organize additional metadata in
 code signing certificates issued by Fulcio instances. This document aims to
 provide a simple directory of values in use with an explanation of their
 meaning.
 
+## Requirements to support signing with CI/CD workload identities
+
+In order to support Sigstore code signing with CI/CD based workfload identities the following claims must be included in the OIDC ID Token. See example claim values for each extension in the detailed [Directory](#directory). 
+
+Requirements:
+
+- MUST include `iss` claim for `Issuer` extension.
+- MUST include claim to support: `Build Signer URI` that identifies the specific build instructions that are responsible for signing.
+- MUST include claim to support: `Build Signer Digest` which is an immutable reference to a specific version of the build instructions that are responsible for signing.
+- MUST include claim to support: `Runner Environment` that differentiates between builds that took place in platform-hosted cloud infrastructure or customer-hosted infrastructure.
+
+Recommended:
+
+- SHOULD include `iss` that uniquely identifies ID tokens originating from the CI/CD system, e.g. not shared with OIDC OAuth 2.0 tokens for email/username logins.
+- SHOULD include claim to support: `Source Repository URI`
+- SHOULD include claim to support: `Source Repository Digest`
+- SHOULD include claim to support: `Source Repository Ref`
+- SHOULD include claim to support: `Source Repository Identifier`
+- SHOULD include claim to support: `Source Repository Owner URI`
+- SHOULD include claim to support: `Source Repository Owner Identifier`
+
+Nice-to-haves:
+
+- MAY include claim to support: `Build Config URI`
+- MAY include claim to support: `Build Config Digest`
+- MAY include claim to support: `Build Trigger`
+- MAY include claim to support: `Run Invocation URI`
+
+## Terminology
+
+- `Identifier`: Immutable resource identifier, e.g. uuid/primary key ID
+- `URI`: SHOULD be a fully qualified URL when available. MAY be a mutable resource identifier, e.g. `https://scm.com/example/repository`
+  - Fully qualified URL: Complete URL with protocol.
+- `Digest`: Immutable source repository reference, e.g. git commit SHA
```

Comment:
nit: I assume a digest isn't always a source repository ref, like for "build config digest", so should this instead say "Output of a cryptographic hash function"